### PR TITLE
One facet per module checks and add facet action update.

### LINKED ISSUE #2292: User reported error: only one facet allowed
```
Only one 'Google App Engine Flexible' facet per module allowed, but 2 facets found in module
Stack trace:
java.lang.Throwable: Only one 'Google App Engine Flexible' facet per module allowed, but 2 facets found in module
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:123)
	at com.intellij.facet.FacetManagerImpl.commit(FacetManagerImpl.java:469)
	at com.intellij.facet.FacetManagerImpl.commit(FacetManagerImpl.java:381)
	at com.intellij.facet.impl.FacetModelImpl.commit(FacetModelImpl.java:96)
	at com.intellij.facet.FacetManagerImpl.addFacet(FacetManagerImpl.java:146)
	at com.google.cloud.tools.intellij.appengine.java.facet.flexible.AppEngineFlexibleSupportProvider$AppEngineFlexibleSupportConfigurable.addSupport(AppEngineFlexibleSupportProvider.java:188)
	at com.google.cloud.tools.intellij.appengine.java.facet.AddAppEngineFrameworkSupportDialog$1.run(AddAppEngineFrameworkSupportDialog.java:70)
	at com.intellij.openapi.application.RunResult.run(RunResult.java:35)
	at com.intellij.openapi.application.WriteAction.execute(WriteAction.java:37)
	at com.google.cloud.tools.intellij.appengine.java.facet.AddAppEngineFrameworkSupportDialog.doOKAction(AddAppEngineFrameworkSupportDialog.java:73)
	at com.intellij.openapi.ui.DialogWrapper$OkAction.doAction(DialogWrapper.java:1868)
	at com.intellij.openapi.ui.DialogWrapper$DialogWrapperAction.actionPerformed(DialogWrapper.java:1828)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2348)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.DefaultButtonModel.setPressed(DefaultButtonModel.java:259)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.Component.processMouseEvent(Component.java:6548)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3325)
	at java.awt.Component.processEvent(Component.java:6313)
```

plugin 18.10.1
idea 2018.1.6

### PR DESCRIPTION
**Pull request by ivanporty**
Fixes #2292.

Fixed add facets action to use more complete check to make sure a module doesn't have App Engine facets added already, also added a check into add facet for App Engine flexible code where exception tends to happen in case there are other paths where adding facets are possible. Tested on standard and flexible projects, confirmed to fix the issue.

### CODE REVIEW

File: app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/flexible/AppEngineFlexibleSupportProvider.java [hunk 1]
```diff
@@ -183,6 +184,12 @@ public void addSupport(
         @NotNull ModifiableModelsProvider modifiableModelsProvider) {
       FacetType<AppEngineFlexibleFacet, AppEngineFlexibleFacetConfiguration> facetType =
           AppEngineFlexibleFacet.getFacetType();
+      // pre-condition - multiple facets of the same type are not allowed and result in a critical
+      // error, check for existing facet explicitly and do nothing if it was already added.
+      if (AppEngineUtil.isAnyAppEngineFacetAlreadyAdded(module)) {
```
Comment (etanshaul) [inline app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/flexible/AppEngineFlexibleSupportProvider.java]: this checks for multiple facets of different types right? This contradicts the comment I think.

Comment (ivanporty) [reply app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/flexible/AppEngineFlexibleSupportProvider.java]: Right, thanks, I can fix the comment. For now, the behavior is we don't allow more than one facet, and we cannot combine standard and flex, is this correct from App Engine point of view?

Comment (etanshaul) [reply app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/flexible/AppEngineFlexibleSupportProvider.java]: From our plugin's point of view we only want to allow a single facet (despite it being possible to have an app configured for both flex and standard).

Comment (ivanporty) [reply app-engine/java/src/com/google/cloud/tools/intellij/appengine/java/facet/flexible/AppEngineFlexibleSupportProvider.java]: Awesome, thanks.

**Comment by ivanporty** [comment]
done.

**Comment by ivanporty** [comment]
@etanshaul should be ready to go.

**Pull request merged**